src/app/model/CheckmateHasOccurred.ts

```diff
@@ -0,0 +1,8 @@
+import { King } from './pieces';
+import { Square } from './Types';
+
+export type CheckmateHasOccurred = {
+  eventType: 'CheckmateHasOccurred';
+  king: King;
+  onSquare: Square | undefined;
```

Comment:
Chyba nie może być checkmate bez oznaczonego square?

Reply:
tutaj podbijam, jeśli szach mat to musi powiedzieć na jakiej pozycji :P